src/processes/cluster.jl

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+   ClusterProcess(p, ofun)
+
+A cluster process with parent process `p` and offsprings generated
+with `ofun`. It is a function that take a parent point and generates
+a set of offspring points as a PointSet.
+"""
+struct ClusterProcess{P<:PointProcess,F<:Function} <: PointProcess
+  p::P
+  ofun::F
+end
+
+function ClusterProcess(p::PointProcess, o::PointProcess, gfun::Function)
+  ClusterProcess(p, parent -> rand(o, gfun(parent)))
+end
+
+function ClusterProcess(p::PointProcess, o::PoissonProcess{<:Function}, gfun::Function; centered = true)
+  if centered
+    ClusterProcess(p, parent -> rand(PoissonProcess(x -> o.λ(Point(x - parent))), gfun(parent)))
+  else
+    ClusterProcess(p, parent -> rand(o, gfun(parent)))
+  end
+end
```

Comment:
Can you elaborate on this method? Isn't it a bit redundant compared with the previous method? Why the special case with Poisson? Can we generalize this centralization feature somehow or leave it as a helper function?

Also, I am always scared of Function fields because people can pass in anything they want and get very scary error messages. We should be careful in the docstring if we really want to preserve this concept of `ofun` and `gfun`. It would be nice to add examples to the docstring in a `## Examples` section.